docs/source/guides/inference.mdx

```diff
@@ -1,58 +1,314 @@
+    - [x] [`~InferenceClient.automatic_speech_recognition`]
+    - [x] [`~InferenceClient.text_to_speech`]
+- computer vision
+    - [x] [`~InferenceClient.image_classification`]
+    - [x] [`~InferenceClient.image_segmentation`]
+    - [x] [`~InferenceClient.image_to_image`]
+    - [x] [`~InferenceClient.image_to_text`]
+    - [ ] object detection
+    - [x] [`~InferenceClient.text_to_image`]
+- multimodal
+    - [ ] documentation question answering
+    - [ ] visual question answering
+- nlp
+    - [x] [`~InferenceClient.conversational`]
+    - [x] [`~InferenceClient.feature_extraction`]
+    - [ ] fill mask
+    - [ ] question answering
+    - [x] [`~InferenceClient.sentence_similarity`]
+    - [x] [`~InferenceClient.summarization`]
+    - [ ] table question answering
+    - [ ] text classification
+    - [ ] text generation
+    - [ ] token classification
+    - [ ] translation
+    - [ ] zero shot classification
+- tabular
+    - [ ] tabular classification
+    - [ ] tabular regression
+
+This list is meant to be completed to support all tasks. If you are particularly interested in a task not yet supported,
+please let us know on [huggingface_hub's repo](https://github.com/huggingface/huggingface_hub).
+
+<Tip>
+
+Check out the tasks page (https://huggingface.co/tasks) to learn more about each task, how to use them and what are the
+most popular models for each of them.
+
+</Tip>
+
+<Tip warning={true}>
+
+The API is designed to be simple. Not all parameters and options are available or described for the end user. Check out
+[this page](https://huggingface.co/docs/api-inference/detailed_parameters) if you are interested in learning more about
+all the parameters available for each task.
+
+</Tip>
+
+## Custom requests
+
+Despite the simplicity of the [`InferenceClient`] API, it is not always possible to cover all use cases. The
+[`InferenceClient.post`] method allows you to send any request to the Inference API. With this flexibility, you can
+for example handle yourself the parsing of the inputs and output. 
+
+```python
+>>> from huggingface_hub import InferenceClient
+>>> client = InferenceClient()
+>>> response = client.post(json={"inputs": "An astronaut riding a horse on the moon."}, model="stabilityai/stable-diffusion-2-1")
+>>> response.content # raw bytes
+b'...'
 ```
 
-The metadata in the model card and configuration files (see [here](https://huggingface.co/docs/hub/models-widgets#enabling-a-widget) for more details) determines the pipeline type. For example, when using the [bert-base-uncased](https://huggingface.co/bert-base-uncased) model, the Inference API can automatically infer that this model should be used for a `fill-mask` task.
+In the example above, the generated image is returned as raw bytes. This can be helpful if you don't have `Pillow`
+installed in your setup and just care about the binary content of the image. [`InferenceClient.post`] is also useful
+to handle tasks that are not yet officially supported.
```

Comment:
Maybe move this to the intro of the Custom requests section. Feels a bit off to introduce/mention this at the end.